plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/split/HudiBackgroundSplitLoader.java

```diff
@@ -149,4 +174,45 @@ private void generateSplits(boolean useIndex)
             throw new TrinoException(HUDI_CANNOT_OPEN_SPLIT, "Error generating Hudi split", e);
         }
     }
+
+    private Map<String, Partition> getPartitionInfos()
+    {
+        if (!enableMetadataTable || !isMetadataPartitionListingEnabled) {
+            return lazyPartitionMap.get();
+        }
+
+        try {
+            PartitionValueExtractor partitionValueExtractor = getPartitionValueExtractor(lazyMetaClient.get().getTableConfig());
+            List<HiveColumnHandle> hivePartitionColumns = tableHandle.getPartitionColumns();
+
+            List<Column> partitionColumns = hivePartitionColumns.stream()
+                    .map(column -> new Column(
+                            column.getName(),
+                            column.getHiveType(),
+                            column.getComment(),
+                            Collections.emptyMap()))
+                    .toList();
+            log.info("Listing partitions for %s.%s via metadata table using partition value extractor %s",
+                    tableHandle.getSchemaName(), tableHandle.getTableName(), partitionValueExtractor.getClass().getSimpleName());
+            Map<String, Partition> metadataPartitions = lazyTableMetadata.get()
+                    .getAllPartitionPaths().stream()
+                    .map(partitionPath -> HudiUtil.buildPartition(partitionPath, partitionColumns, tableHandle, partitionValueExtractor))
```

Comment:
Let's import the static methods directly
```suggestion
                    .map(partitionPath -> buildPartition(partitionPath, partitionColumns, tableHandle, partitionValueExtractor))
```